pkg/resourcegenerator/resourceutils/helpers.go

```diff
@@ -18,24 +21,42 @@ func ShouldScaleToZero(jsonReplicas *apiextensionsv1.JSON) bool {
 	return false
 }
 
+// MatchesRegex checks if a string matches a regexp
+func matchesRegex(s string, pattern string) bool {
+	obj, err := regexp.Match(pattern, []byte(s))
+	return obj && err == nil
+}
+
+// GetImageVersion returns the version part of an image string
 func GetImageVersion(imageVersionString string) string {
-	parts := strings.Split(imageVersionString, ":")
+	// Find position of first "@", remove it and everything after it
+	if strings.Contains(imageVersionString, "@") {
+		imageVersionString = strings.Split(imageVersionString, "@")[0]
+		imageVersionString = imageVersionString + ":unknown"
+	}
 
-	// Implicitly assume version "latest" if no version is specified
-	if len(parts) < 2 {
+	// If no version is given, assume "latest"
+	if !strings.Contains(imageVersionString, ":") {
 		return "latest"
 	}
 
+	// Split image string into parts
+	parts := strings.Split(imageVersionString, ":")
+
 	versionPart := parts[1]
```

Comment:
Handles `registry:5000/foo/bar:1.2.3`

```suggestion
	lastColonPos := strings.LastIndex(imageVersionString, ":")
	if lastColonPos == -1 || lastColonPos == len(imageVersionString)-1 {
		return "latest"
	}

	versionPart := imageVersionString[lastColonPos+1:]
	imageVersionString = imageVersionString[:lastColonPos]

	// Replace "+" with "-" in version text if version includes one
	versionPart = strings.ReplaceAll(versionPart, "+", "-")
```